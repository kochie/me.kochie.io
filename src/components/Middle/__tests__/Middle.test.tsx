import * as renderer from 'react-test-renderer'
import * as React from 'react'
import { jest } from '@jest/globals'

describe('Middle Component', () => {
  const MOCK_DATA = {
    items: [
      {
        id: 'https://blog.kochie.io/articles/02-contact-tracing',
        url: 'https://blog.kochie.io/articles/02-contact-tracing',
        title: 'Building a Contact Tracing Platform',
        summary:
          "Learn how to automate and build an application to trace exposures using Amazon's Cloud Development Kit (CDK)",
        image:
          'https://blog.kochie.io/articles-1/02-contact-tracing/omar-flores-MOO6k3RaiwE-unsplash.jpg',
        date_modified: '2021-08-22T23:00:00.000Z',
        author: {
          name: 'Robert Koch',
          url: 'https://blog.kochie.io/authors/kochie',
        },
      },
    ],
  }

  beforeAll(async () => {
    jest.unstable_mockModule('swr', () => {
      // console.log('mock')
      return {
        default: jest.fn(() => ({
          data: MOCK_DATA,
          error: null,
        })),
      }
    })

    await import('swr')
  })

  it('middle renders correctly', async () => {
    const Middle = await import('../Middle')
    const tree = renderer.create(<Middle.default />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
