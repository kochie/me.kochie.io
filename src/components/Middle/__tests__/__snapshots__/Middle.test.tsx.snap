// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middle renders correctly 1`] = `
<div
  className="py-24 px-24 inline-grid bg-gray-200 dark:bg-gray-800 transform-gpu transition duration-700 prose dark:prose-light max-w-none"
>
  <div
    className=""
  >
    <h1>
      I’m Rob, an engineer from Melbourne, Australia.
    </h1>
    <p>
      I'm a graduate of Monash University where I studied Electrical Engineering, Computer Science, and Mathematics. While studying I was involved with the IT society 
      <a
        href="http://wired.org.au"
      >
        wired
      </a>
      , where I held the title of President in 2017.
    </p>
    <p>
      I currently work as a Cloud Architect for 
      <a
        href="https://aws.amazon.com"
      >
        Amazon Web Services
      </a>
      , where I build solutions for customers in the public sector, think universities, government departments, and non-for-profits. 
    </p>
    <p>
      Before that I worked as a Software Engineer at 
      <a
        href="https://localz.com"
      >
        Localz
      </a>
       who are providing last mile solutions for people around the world.
    </p>
    <p>
      I also helped run the 
      <a
        href="https://unihack.net"
      >
        Unihack
      </a>
       organisation in Melbourne, where we ran hackathons for students around Australia. 
    </p>
    <p>
      I'm an avid runner and enjoy bouldering all around Melbourne and beyond.
    </p>
    <p>
      Apart from that, I love gaming, karate, bike riding, and electronics and try to do at least one of these things every day. I’m always looking for something new and interesting to do, so feel free to browse my portfolio or view my resume if you want to see some of the cool things I’ve done.
    </p>
  </div>
  <hr
    className="divider"
  />
  <div
    className="button"
  >
    <a
      download="Robert_Koch.pdf"
      href="/Robert_Koch.pdf"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "margin": "auto",
          "textDecoration": "none",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-file-user fa-w-12 fa-1x "
        data-icon="file-user"
        data-prefix="fad"
        focusable="false"
        role="img"
        style={
          Object {
            "--fa-primary-color": "white",
            "--fa-secondary-color": "#ffdd00",
            "--fa-secondary-opacity": "1",
            "stroke": "black",
            "strokeWidth": "20px",
          }
        }
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          className="fa-group"
          style={Object {}}
        >
          <path
            className="fa-secondary"
            d="M384 128H272a16 16 0 0 1-16-16V0H24A23.94 23.94 0 0 0 0 23.88V488a23.94 23.94 0 0 0 23.88 24H360a23.94 23.94 0 0 0 24-23.88V128zm-192 64a64 64 0 1 1-64 64 64 64 0 0 1 64-64zm112 236.8c0 10.61-10 19.2-22.4 19.2H102.4C90 448 80 439.4 80 428.8v-19.2c0-31.81 30.09-57.6 67.1-57.6h5a103.22 103.22 0 0 0 79.7 0h5c37.11 0 67.2 25.79 67.2 57.6z"
            fill="currentColor"
            style={Object {}}
          />
          <path
            className="fa-primary"
            d="M377 105L279.1 7a24 24 0 0 0-17-7H256v112a16 16 0 0 0 16 16h112v-6.1a23.9 23.9 0 0 0-7-16.9zM192 320a64 64 0 1 0-64-64 64 64 0 0 0 64 64zm44.8 32h-5a103.22 103.22 0 0 1-79.7 0h-5c-37 0-67.1 25.79-67.1 57.6v19.2c0 10.6 10 19.2 22.4 19.2h179.2c12.37 0 22.4-8.59 22.4-19.2v-19.2c0-31.81-30.09-57.6-67.2-57.6z"
            fill="currentColor"
            style={Object {}}
          />
        </g>
      </svg>
      <span
        style={
          Object {
            "color": "white",
            "fontSize": "25px",
            "letterSpacing": "-0.6px",
            "margin": "0 10px",
          }
        }
      >
        Download Resume
      </span>
    </a>
  </div>
  <hr
    className="divider"
  />
  <div
    className=""
  >
    <div>
      <h2>
        Recent Work
      </h2>
      <div
        className="grid 2xl:grid-cols-3 md:grid-cols-2 grid-cols-1 grid-flow-row gap-10 auto-cols-max"
      >
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://kochie.space"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Space"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fspace.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Fspace.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Fspace.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Fspace.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Fspace.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Fspace.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Space"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://kochie.space"
              >
                Space
              </a>
            </h3>
            <p
              className="text-sm"
            >
              A small webapp that follows the mouse.
            </p>
          </div>
        </div>
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://3body.now.sh"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Particlez"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fparticlez.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Fparticlez.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Fparticlez.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Fparticlez.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Fparticlez.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Fparticlez.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Particlez"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://3body.now.sh"
              >
                Particlez
              </a>
            </h3>
            <p
              className="text-sm"
            >
              A real time n-body particle simulator for the web.
            </p>
          </div>
        </div>
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://github.com/kochie/guardian-server"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Guardian"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fguardian.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Fguardian.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Fguardian.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Fguardian.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Fguardian.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Fguardian.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Guardian"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://github.com/kochie/guardian-server"
              >
                Guardian
              </a>
            </h3>
            <p
              className="text-sm"
            >
              An experiment in transferring cyptographic keys between devices without the user needing to copy keys themselves.
            </p>
          </div>
        </div>
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://bitbucket.org/kochie/fyp"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Optimising Neural Networks for Embedded Systems"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Ftensorflow.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Ftensorflow.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Ftensorflow.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Ftensorflow.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Ftensorflow.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Ftensorflow.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Optimising Neural Networks for Embedded Systems"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://bitbucket.org/kochie/fyp"
              >
                Optimising Neural Networks for Embedded Systems
              </a>
            </h3>
            <p
              className="text-sm"
            >
              My final year project - focusing on modifying the activation layers of convolutional neural networks to work better on embedded devices.
            </p>
          </div>
        </div>
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://github.com/kochie/juice"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Juice"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fjuice.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Fjuice.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Fjuice.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Fjuice.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Fjuice.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Fjuice.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Juice"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://github.com/kochie/juice"
              >
                Juice
              </a>
            </h3>
            <p
              className="text-sm"
            >
              A Json-Web-Token analyser.
            </p>
          </div>
        </div>
        <div
          className="group rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 hover:shadow-2xl shadow-xl transition duration-300"
        >
          <div
            className="relative rounded-t-xl overflow-hidden"
            style={
              Object {
                "padding": "30% 0",
              }
            }
          >
            <a
              href="https://github.com/kochie/mcgurrin"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="McGurrin"
                    className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fkeyboard.webp&w=1200&q=75"
                    srcSet="/_next/image?url=%2Fimages%2Fkeyboard.webp&w=320&q=75 320w, /_next/image?url=%2Fimages%2Fkeyboard.webp&w=420&q=75 420w, /_next/image?url=%2Fimages%2Fkeyboard.webp&w=768&q=75 768w, /_next/image?url=%2Fimages%2Fkeyboard.webp&w=1024&q=75 1024w, /_next/image?url=%2Fimages%2Fkeyboard.webp&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="McGurrin"
                  className="absolute top-0 left-0 w-full h-full group-hover:scale-110 transform-gpu transition duration-100 filter grayscale-35 group-hover:grayscale-0"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="px-5"
          >
            <h3
              className="text-lg mt-0"
            >
              <a
                href="https://github.com/kochie/mcgurrin"
              >
                McGurrin
              </a>
            </h3>
            <p
              className="text-sm"
            >
              A tool to help with learning to touch type.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="divider"
  />
  <div
    className=""
  >
    <div>
      <h2
        className="text-xl mb-6"
      >
        Recent Articles
      </h2>
      <div
        className="flex flex-col gap-10"
      >
        <div
          className="group flex rounded-xl bg-cool-gray-300 dark:bg-cool-gray-700 shadow-2xl"
        >
          <div
            className="overflow-hidden rounded-l-xl flex-none relative h-40 w-80"
          >
            <a
              href="https://blog.kochie.io/articles/02-contact-tracing"
            >
              <div
                style={
                  Object {
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "left": 0,
                    "margin": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <noscript>
                  <img
                    alt="Building a Contact Tracing Platform"
                    className="group-hover:scale-110 transition transform-gpu duration-100"
                    data-nimg={true}
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1200&q=75"
                    srcSet="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=320&q=75 320w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=420&q=75 420w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=768&q=75 768w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1024&q=75 1024w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Farticles-1%2F02-contact-tracing%2Fomar-flores-MOO6k3RaiwE-unsplash.jpg&w=1200&q=75 1200w"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "cover",
                        "objectPosition": "center",
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                </noscript>
                <img
                  alt="Building a Contact Tracing Platform"
                  className="group-hover:scale-110 transition transform-gpu duration-100"
                  data-nimg={true}
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                    Object {
                      "backgroundImage": "url(\\"data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgMCAgMDAwMEAwMEBQgFBQQEBQoHBwYIDAoMDAsKCwsNDhIQDQ4RDgsLEBYQERMUFRUVDA8XGBYUGBIUFRT/2wBDAQMEBAUEBQkFBQkUDQsNFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBT/wAARCAAKABIDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAYEBQcI/8QAKBAAAQMDAwMDBQAAAAAAAAAAAQIDBBEFABIhBgcjMRQiEzJCUXGR/8QAFwEAAwEAAAAAAAAAAAAAAAAAAAIEBf/EABwRAAEEAwEAAAAAAAAAAAAAAAEAQQIhE9HwA//aAAwDAQACEQMRAD8AaLZceON9VUT4txmRum1FMszi6ta3wNtKSRXcmtCk/UP3lnJ5d0yb6d8kXeEwWJchC2rEw++4j1CiSEgJAAVt7qHb20ziPgFwlVvSfUvaWk1QPkNEdsePxk2Dc5l4F4M+W/OKEVSZLhc00QkClSabE/3FHnKJMqdloEihbOtEcfiuOLWu4nWo1PeT5wxFXHa1q7aPJ+0YZPjHE7RkPAaX/9k=\\")",
                      "backgroundPosition": "center",
                      "backgroundSize": "cover",
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "filter": "blur(20px)",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </a>
          </div>
          <div
            className="ml-5"
          >
            <h3
              className="hover:underline"
            >
              <a
                href="https://blog.kochie.io/articles/02-contact-tracing"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Building a Contact Tracing Platform
              </a>
            </h3>
            <p>
              Learn how to automate and build an application to trace exposures using Amazon's Cloud Development Kit (CDK)
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
